pulsar/internal/service_name_resolver.go

```diff
@@ -98,8 +106,9 @@ func (r *pulsarServiceNameResolver) UpdateServiceURL(u *url.URL) error {
 	r.AddressList = addresses
 	r.ServiceURL = u
 	r.ServiceURI = uri
-	rand.Seed(time.Now().Unix()) // initialize global pseudo random generator
-	atomic.StoreInt32(&r.CurrentIndex, int32(rand.Intn(len(addresses))))
+	r.mutex.Lock()
```

Comment:
We should move the
```
r.mutex.Lock()
defer r.mutex.Unlock()
```
to under `uri, err := NewPulsarServiceURIFromURL(u)`
Since state other piece of state are being changed.

Reply:
In this PR, I only mentioned `CurrentIndex`. However I think your comment is correct, so I'll address your comment.